jiant/tasks/lm.py

```diff
@@ -172,3 +175,102 @@ def __init__(self, path, *args, **kw):
             "val": os.path.join(path, "valid.sentences.txt"),
             "test": os.path.join(path, "test.sentences.txt"),
         }
+
+
+class MaskedLanguageModelingTask(LanguageModelingTask):
+    """
+       Generic Masked Language Modeling Task
+    """
+
+    pass
+
+
+@register_task("mlm", rel_path="WikiText103/")
+class MLMTask(MaskedLanguageModelingTask):
+    """
+    Masked language modeling task on Toronto Books dataset
+    Attributes:
+        max_seq_len: (int) maximum sequence length
+        min_seq_len: (int) minimum sequence length
+        files_by_split: (dict) files for three data split (train, val, test)
+    """
+
+    def __init__(self, path, *args, **kw):
+        super().__init__(path, *args, **kw)
+        self.files_by_split = {
+            "train": os.path.join(path, "train.sentences.txt"),
+            "val": os.path.join(path, "valid.sentences.txt"),
+            "test": os.path.join(path, "test.sentences.txt"),
+        }
+
+    def update_metrics(self, out, batch=None):
+        # self.scorer1(logits,labels)
+        self.scorer1(out["loss"].mean())
+        return
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text
+        Args:
+            path: (str) data file path
+        """
+        import csv
+
+        f = open(path, "r")
+        reader = csv.reader(f)
+        text = list(reader)
+        for i in range(len(text)):
+            row = text[i]
+            toks = "".join(row)
+            yield tokenize_and_truncate(self._tokenizer_name, toks, self.max_seq_len)
+
+    def process_split(
+        self, split, indexers, model_preprocessing_interface
+    ) -> Iterable[Type[Instance]]:
+        """Process a language modeling split by indexing and creating fields.
+        Args:
+            split: (list) a single list of sentences
+            indexers: (Indexer object) indexer to index input words
+        """
+
+        def _make_instance(sent_):
+            """ Forward targs adds <s> as a target for input </s>
+            and bwd targs adds </s> as a target for input <s>
+            to avoid issues with needing to strip extra tokens
+            in the input for each direction """
+            sent_ = model_preprocessing_interface.boundary_token_fn(sent_)  # Add <s> and </s>
+            input_sent = sentence_to_text_field(sent_, indexers)
+            d = {
+                "input": input_sent,
+                "targs": SequenceLabelField(
```

Comment:
are input and targs  the same sequence? do we need two separate domains?

Reply:
Yes, I think so. I don't think they should be two separate domains because the model is supposed to predict the correct original word that was replaced by MASK/a random word/kept as is.